### PR TITLE
Add native video player

### PR DESCRIPTION
**Pull request by thornbill**
Adds a native video player

* [x] Inject web plugin for native player
* [x] Add setting to enable native player
* [x] Disable plugin for server versions < 10.7
* [x] Add device profile to support transcoding
* [x] Fix progress tracking
* [x] Fix playing next item in queue
* [x] Add any necessary functions to the plugin for full compatibility with web
* [x] Force the player to always use fullscreen
* [x] Honor rotation lock setting
* [x] Refresh webview when changing native video player setting

**Limitations with the native video player**

* Rotation lock is not fully supported (screen is rotated initially, but can be rotated back)
* Subtitles must be selected before starting playback to be listed in the native player
* Picture-in-Picture is not supported

### CODE REVIEW

**Comment by thornbill** [comment]
This is ready for review now. SonarCloud is just mad at the duplication in the device profiles. :roll_eyes:

**Comment by anthonylavado** [comment]
@thornbill Oh boy. Okay.

File: components/VideoPlayer.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	return (
+		<Video
+			ref={player}
+			usePoster
+			posterSource={{ uri: rootStore.mediaStore.posterUri }}
+			resizeMode='contain'
+			useNativeControls
+			onReadyForDisplay={() => {
+				player.current?.presentFullscreenPlayer()
+					.catch(console.debug);
```
Comment (heyhippari) [inline components/VideoPlayer.js]: Wouldn't it be good to have some message here for the user? Or is that taken care of by jf-web noticing that playback failed?

**Review by heyhippari — APPROVED** [review]
LGTM, though I can't test it on an actual device.

File: components/VideoPlayer.js (hunk 1 above)
Comment (thornbill) [reply components/VideoPlayer.js]: Good call! Playback errors are handled in the `onError` function, but I added this to catch errors thrown when showing the fullscreen player when it is already being shown. I added some local state to debounce calls to present and dismiss and added an Alert dialog if an error happens when presenting.

**Review by anthonylavado — APPROVED** [review]

**Pull request merged**